grazel-gradle-plugin/src/main/kotlin/com/grab/grazel/migrate/android/AndroidInstrumentationBinaryDataExtractor.kt

```diff
@@ -0,0 +1,136 @@
+        val extension = project.extensions.getByType<BaseExtension>()
+        val deps = projectDependencyGraphs
+            .directDependencies(
+                project,
+                BuildGraphType(ConfigurationScope.ANDROID_TEST, mergedVariant.variant)
+            ).map { dependency ->
+                gradleDependencyToBazelDependency.map(project, dependency, mergedVariant)
+            } +
+            dependenciesDataSource.collectMavenDeps(
+                project,
+                BuildGraphType(ConfigurationScope.ANDROID_TEST, mergedVariant.variant)
+            ) +
+            BazelDependency.ProjectDependency(
+                project,
+                "_lib${mergedVariant.variantName.variantNameSuffix()}"
+            )
+
+        return project.extract(
+            mergedVariant = mergedVariant,
+            extension = extension,
+            deps = deps,
+            sourceSetType = sourceSetType,
+        )
+    }
+
+    private fun Project.extract(
+        mergedVariant: MergedVariant,
+        extension: BaseExtension,
+        deps: List<BazelDependency>,
+        sourceSetType: SourceSetType,
+    ): AndroidInstrumentationBinaryData {
+
+        val migratableSourceSets = mergedVariant.variant.sourceSets
+            .filterIsInstance<AndroidSourceSet>()
+            .toList()
+
+        val packageName = androidManifestParser.parsePackageName(
+            extension,
+            migratableSourceSets
+        ) ?: ""
+
+        val debugKey = keyStoreExtractor.extract(
+            rootProject = rootProject,
+            variant = variantDataSource.getMigratableBuildVariants(this).firstOrNull()
+        )
+
+        val associate = BazelDependency.ProjectDependency(
+            dependencyProject = this,
+            suffix = "_lib${mergedVariant.variantName.variantNameSuffix()}_kt"
+        )
+
+        val resources = unitTestResources(migratableSourceSets.asSequence()).toList()
+        val resourceFiles = androidSources(migratableSourceSets, SourceSetType.RESOURCES).toList()
+
+        val srcs = androidSources(migratableSourceSets, sourceSetType).toList()
+        val testInstrumentationRunner = extension.extractTestInstrumentationRunner()
+
+        return AndroidInstrumentationBinaryData(
+            name = "${name}${mergedVariant.variantName.variantNameSuffix()}-android-test",
+            associates = listOf(associate),
+            customPackage = packageName,
+            debugKey = debugKey,
+            deps = deps,
+            instruments = BazelDependency.StringDependency(
+                ":${name}${mergedVariant.variantName.variantNameSuffix()}"
+            ),
+            resources = resources,
+            resourceFiles = resourceFiles,
+            srcs = srcs,
+            testInstrumentationRunner = testInstrumentationRunner,
+        )
+    }
+}
+
+internal fun BaseExtension.extractTestInstrumentationRunner(): String =
+    defaultConfig.testInstrumentationRunner ?: ""
```

Comment:
Why default to empty string? Can we null it and handle the case in the calling code? Empty string prevents use of elvis operator at call site

Reply:
fixed this in newer commit to ensure it is null and getting handled